diff -up java-10-openjdk-10.0.2.13-8.x86_64/openjdk/src/java.base/share/native/libjimage/imageFile.cpp.omv~ java-10-openjdk-10.0.2.13-8.x86_64/openjdk/src/java.base/share/native/libjimage/imageFile.cpp
--- java-10-openjdk-10.0.2.13-8.x86_64/openjdk/src/java.base/share/native/libjimage/imageFile.cpp.omv~	2019-01-10 01:50:49.778812248 +0100
+++ java-10-openjdk-10.0.2.13-8.x86_64/openjdk/src/java.base/share/native/libjimage/imageFile.cpp	2019-01-10 01:51:56.430101637 +0100
@@ -57,14 +57,16 @@ const char FileSeparator = '/';
 
 // Compute the Perfect Hashing hash code for the supplied UTF-8 string.
 s4 ImageStrings::hash_code(const char* string, s4 seed) {
+    assert(seed > 0 && "invariant");
     // Access bytes as unsigned.
     u1* bytes = (u1*)string;
+    u4 useed = (u4)seed;
     // Compute hash code.
     for (u1 byte = *bytes++; byte; byte = *bytes++) {
-        seed = (seed * HASH_MULTIPLIER) ^ byte;
+        useed = (useed * HASH_MULTIPLIER) ^ byte;
     }
     // Ensure the result is not signed.
-    return seed & 0x7FFFFFFF;
+    return (s4)(useed & 0x7FFFFFFF);
 }
 
 // Match up a string in a perfect hash table.
